packages/api-request-builder/src/create-service.js

```diff
@@ -109,29 +109,26 @@ export default function createService (
     // - `withProjectKey: false`: will omit the projectKey from the URI
     build (uriOptions: UseKey = { withProjectKey: true }): string {
       const { withProjectKey } = uriOptions
-
       const queryParams = buildQueryString(this.params)
-      const version = this.params.version
 
-      const uri =
-        (withProjectKey ? `/${options}` : '') +
-        endpoint +
-        getIdOrKey(this.params) +
-        (queryParams ? `?${queryParams}` : '') +
-        (version ? `?version=${version}` : '')
+      let uri = withProjectKey ? `/${options}` : ''
+      if (this.params.id)
+        uri += `${endpoint}/${this.params.id}` +
+          `${queryParams ? `?${queryParams}` : ''}`
+
+      else if (this.params.key)
+        uri += `${endpoint}/key=${this.params.key}` +
+          `${queryParams ? `?${queryParams}` : ''}`
+
+      else if (this.params.customerId)
+        uri += `${endpoint}/?customerId=${this.params.customerId}` +
```

Comment:
Can't you simply put `customerId` also in the `buildQueryString` function like for `version`?

I would also keep the `getIdOrKey` function, makes this thing easier to read. Thanks 🙏

Reply:
I think that on first glance, it could make the code less understandable because a URI can only have one of `id`, `key` or `customerId` so it makes sense for them to be in the same place

Reply:
The `customerId` is a query string parameter, `id` and `key` are URI parts.